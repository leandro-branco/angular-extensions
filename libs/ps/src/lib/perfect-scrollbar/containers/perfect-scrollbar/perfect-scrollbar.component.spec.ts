import {
  ChangeDetectionStrategy,
  Component,
} from '@angular/core';

import {
  async,
  ComponentFixture,
} from '@angular/core/testing';

import {
  configureTest,
  TestFn,
} from '@ngxma/testing';

import {
  PerfectScrollbarModule as NgxPerfectScrollbarModule,
} from 'ngx-perfect-scrollbar';

import {
  PerfectScrollbarComponent,
} from './perfect-scrollbar.component';


describe('PerfectScrollbarComponent', () => {
  let component: PerfectScrollbarComponent;
  let fixture: ComponentFixture<PerfectScrollbarComponent>;

  beforeEach(async(() => {
    const configure: TestFn = (testBed) => {
      testBed.configureTestingModule({
        declarations: [
          PerfectScrollbarComponent,
        ],
        imports: [
          NgxPerfectScrollbarModule,
        ],
      })
        .overrideComponent(PerfectScrollbarComponent, {
          set: { changeDetection: ChangeDetectionStrategy.Default },
        });
    };

    configureTest(configure).then((testBed) => {
      fixture = testBed.createComponent(PerfectScrollbarComponent);
      component = fixture.componentInstance;
      fixture.detectChanges();
    });
  }));

  it('[@component] should create', () => {
    expect(component).toBeTruthy();
  });

  it('[@component] should match snapshot', () => {
    expect(fixture.nativeElement).toMatchSnapshot();
  });
});

@Component({
  selector: 'xma-test-host',
  template: `
  <xma-perfect-scrollbar>
    <div>PLACED INSIDE</div>
  </xma-perfect-scrollbar>`,
})
class TestHostComponent { }

describe('PerfectScrollbarComponent: host component', () => {
  let component: TestHostComponent;
  let fixture: ComponentFixture<TestHostComponent>;

  beforeEach(async(() => {
    const configure: TestFn = (testBed) => {
      testBed.configureTestingModule({
        declarations: [
          TestHostComponent,
          PerfectScrollbarComponent,
        ],
        imports: [
          NgxPerfectScrollbarModule,
        ],
      });
    };

    configureTest(configure).then((testBed) => {
      fixture = testBed.createComponent(TestHostComponent);
      component = fixture.componentInstance;
      fixture.detectChanges();
    });
  }));

  it('[@component] should create', () => {
    expect(component).toBeTruthy();
  });

  it('[@component] should match snapshot', () => {
    expect(fixture.nativeElement).toMatchSnapshot();
  });
});
