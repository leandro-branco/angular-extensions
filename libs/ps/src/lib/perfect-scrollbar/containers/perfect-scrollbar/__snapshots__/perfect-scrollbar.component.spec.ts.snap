// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PerfectScrollbarComponent [@component] should match snapshot 1`] = `
<div
  class="xma-perfect-scrollbar"
  id="root1"
  ng-version="7.0.2"
>
  <perfect-scrollbar>
    <div
      class="ps"
      ng-reflect-disabled="false"
      style="position: static;"
    >
      <div
        class="ps-content"
      />
      <!--bindings={
  "ng-reflect-ng-if": "false"
}-->
      <div
        class="ps__rail-x"
        style="left: 0px; bottom: 0px;"
      >
        <div
          class="ps__thumb-x"
          style="left: 0px; width: 0px;"
          tabindex="0"
        />
      </div>
      <div
        class="ps__rail-y"
        style="top: 0px; right: 0px;"
      >
        <div
          class="ps__thumb-y"
          style="top: 0px; height: 0px;"
          tabindex="0"
        />
      </div>
    </div>
  </perfect-scrollbar>
</div>
`;

exports[`PerfectScrollbarComponent: host component [@component] should match snapshot 1`] = `
<div
  id="root3"
  ng-version="7.0.2"
>
  <xma-perfect-scrollbar
    class="xma-perfect-scrollbar"
  >
    <perfect-scrollbar>
      <div
        class="ps"
        ng-reflect-disabled="false"
        style="position: static;"
      >
        <div
          class="ps-content"
        >
          <div>
            PLACED INSIDE
          </div>
        </div>
        <!--bindings={
  "ng-reflect-ng-if": "false"
}-->
        <div
          class="ps__rail-x"
          style="left: 0px; bottom: 0px;"
        >
          <div
            class="ps__thumb-x"
            style="left: 0px; width: 0px;"
            tabindex="0"
          />
        </div>
        <div
          class="ps__rail-y"
          style="top: 0px; right: 0px;"
        >
          <div
            class="ps__thumb-y"
            style="top: 0px; height: 0px;"
            tabindex="0"
          />
        </div>
      </div>
    </perfect-scrollbar>
  </xma-perfect-scrollbar>
</div>
`;
